### ISSUE TITLE
Help: Support for Windows

### ISSUE THREAD

**Issue by TeoDev1611** [williamboman/nvim-lsp-installer#40]
I use native lsp on Windows and Linux but I try some plugins which work only Linux and MacOS this plugin work in Windows or not ?🤔

**Comment by williamboman** [comment]
Hey @TeoDev1611! None of these installers have been tested on Windows, so YMMV. There are some installers that explicitly does not support Windows, for example [rust_analyzer](https://github.com/williamboman/nvim-lsp-installer/blob/main/lua/nvim-lsp-installer/servers/rust_analyzer/install.mjs#L12-L14). LSP servers that only require you to install an npm package might work though! Have you tried it yet?

**Comment by williamboman** [comment]
Oh, PRs are also very welcomed should you want to add support for Windows!

**Comment by TeoDev1611** [comment]
I I'm trying this now. This not work but in windows not exists /bin/bash this is the error ☹️

**Comment by TeoDev1611** [comment]
> Oh, PRs are also very welcomed should you want to add support for Windows!

I check if I can make this, thanks for your awesome work

**Comment by JonTofant** [comment]
I get the same error on windows /bin/bash

**Comment by williamboman** [comment]
Yeah the current installers rely on Unix shell (all installers have their entrypoint in a bash shell). I imagine some would be really easy to get to work with Windows - the ones that leverage existing package management tools such as [`npm`](https://github.com/williamboman/nvim-lsp-installer/blob/main/lua/nvim-lsp-installer/installers/npm.lua) and [`pip`](https://github.com/williamboman/nvim-lsp-installer/blob/main/lua/nvim-lsp-installer/installers/pip3.lua).

I have barely any experience with Windows - is it common practice nowadays to run things in WSL? Or do people use Powershell?

**Comment by JonTofant** [comment]
I personaly use Powershell.  I agree with you with npm and pip there shouldnt be much of a diffrence

**Comment by TeoDev1611** [comment]
I use powershell or cmd or for run any command in cmd you can use:
```
cmd /c npm --version 
```
For example

**Comment by barungh** [comment]
### Windows 10  
In my Windows machine I am getting this error  

```
[nvim-lsp-installer] Failed to install cssls.
"...r\\start\\nvim-lsp-installer\\lua\\nvim-lsp-installer\\fs.lua:8: Refusing to operate on path outside of the servers root dir
```

**Comment by barungh** [comment]
> Hey @TeoDev1611! None of these installers have been tested on Windows, so YMMV. There are some installers that explicitly does not support Windows, for example [rust_analyzer](https://github.com/williamboman/nvim-lsp-installer/blob/main/lua/nvim-lsp-installer/servers/rust_analyzer/install.mjs#L12-L14). LSP servers that only require you to install an npm package might work though! Have you tried it yet?

rust_analyzer works fine in Windows

Microsoft is consistently working to improve Powershell

I work on both Linux and Windows

Since the release neovim 0.5 ( Lua ) I am trying to set up everything on my Windows machine

with [Coc-nvim](https://github.com/neoclide/coc.nvim) everything was working fine in Powershell 

but I am trying to configure everything with Lua

I tried many configuration like NvChad, LunarVim, VapourVim etc. but every config misses something

Especially I am finding difficulties to install language servers, 
but manually installing them works

with [Vapour/Nvim](https://github.com/hackorum/VapourNvim/) almost all is set like flutter-tool and rust-tool,
but my HTML, CSS servers are not configured properly

**Comment by barungh** [comment]
Most of the servers are installing now,
but I am yet to configure them with lspconfig

got error with few lsp
![image](https://user-images.githubusercontent.com/12285528/131211475-8b6b36ae-5127-4d9a-b520-c70aa696c0b4.png)

**Comment by williamboman** [comment]
> Most of the servers are installing now,
> but I am yet to configure them with lspconfig

Sweet :+1:. If you just want to quickly try it out you can just run a one-off command like so:

```vim
:lua require("nvim-lsp-installer").on_server_ready(function(server) server:setup(); vim.cmd [[ do User LspAttachBuffers ]] end)
```

> got error with few lsp

Yeah not all servers will be supported initially (those with custom install scripts, currently targeting *NIX shell). I'll put in a friendlier error message for Windows! I'm currently looking into whether using Node.js (via [zx](https://github.com/google/zx)) as an intermediary runtime could make cross-platform support easier. So far I got rust_analyzer working, although a bit hacky.

**Comment by williamboman** [comment]
I've not been able to spend more time on this, so I went ahead and merged #70. A majority of LSPs now support Windows. There are a few remaining that require a custom install script which we'll have to deal with on a case by case basis. PRs much welomed!
